azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -59,95 +64,152 @@ def __init__(self, wire_client, full_goal_state=False, base_incarnation=None):
         """
         uri = GOAL_STATE_URI.format(wire_client.get_endpoint())
         self.xml_text = wire_client.fetch_config(uri, wire_client.get_header())
-        xml_doc = parse_doc(self.xml_text)
+        self._xml_doc = parse_doc(self.xml_text)
+
+        self._ext_config_retriever = ext_config_retriever
+        self._wire_client = wire_client
+
+        self.ext_conf = None
+        self.hosting_env = None
+        self.shared_conf = None
+        self.certs = None
+        self.remote_access = None
+        self.artifacts_profile_blob_url = None
```

Comment:
what is the motivation of moving the artifacts profile and status blobs from the ExtensionConfig instance to the GoalState instance?

Seems to me like the extension config retriever should take care or retrieving them and populating the extension config accordingly

Reply:
We can't retrieve the ext_config without them. The ext_config may now come from either Fabric or FastTrack. However, to retrieve the FastTrack ext_config we need to know the artifacts_profile url. It's a chicken and egg scenario.
The status url I moved to be consistent.

Reply:
Yes, we need them to retrieve  the fast track extensions config, but that does not require moving them to the Goal State class. Since they are properties of the fabric extensions config, can they stay in that class?
From now on, it seems like we will always need to retrieve the fabric extension config when we get a new goal state; can that extension config stay as a property in the Goal State class as it is currently?
Today these classes and their properties match 1:1 with the network requests and the files we save to the history folder, and that simplifies debugging and understanding the code. It'd be great if we can keep them as they are today.

Reply:
Will be moving them out of goalstate with the latest changes. Most likely they'll go in the retriever.